batavia/core/builtins.js

```diff
@@ -995,8 +1002,26 @@ batavia.builtins.tuple = function(args) {
     return new batavia.core.Tuple(args[0]);
 };
 
-batavia.builtins.type = function() {
-    throw new batavia.builtins.NotImplementedError("Builtin Batavia function 'type' not implemented");
+batavia.builtins.type = function(args) {
+    var type = 'NoneType'
```

Comment:
At the top of builtins.js, there's a template for builtin methods - they should take (args, kwargs) as arguments, and do some light validation of those arguments to make sure you're not passing in too many/not enough arguments. There's also a `__doc__`  attribute that should be added.